### PR TITLE
Improve Registry

### PR DESCRIPTION
**Pull request by williambl**
Registry implementing Iterable mirrors how modern vanilla Registry does. It's useful for those of us who need to do something to every registered Tile/ItemType. I also added values(), ids(), and serialisedIds() functions.

### CODE REVIEW

File: src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java [hunk 1]
```diff
@@ -248,4 +246,44 @@ protected void onRemap(T remappedValue, int newSerialisedId) {
...
+	 *
+	 * @return the ids of all objects stored in this registry.
+	 */
+	public Set<Id> ids() {
+		return this.byRegistryId.keySet();
+	}
+
+	/**
+	 *
+	 * @return the serialised ids of all objects stored in this registry
```
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java]: Can we have a full stop at the end of this sentence?

File: src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java [hunk 2]
```diff
@@ -248,4 +246,44 @@ protected void onRemap(T remappedValue, int newSerialisedId) {
...
+	 */
+	public Set<Integer> serialisedIds() {
+		return this.bySerialisedId.keySet();
+	}
+
+	@Override
+	@Nonnull
+	public Iterator<T> iterator() {
+		return values().iterator();
```
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java]: Can you reference values() with `this.`

File: src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java [hunk 3]
```diff
@@ -248,4 +246,44 @@ protected void onRemap(T remappedValue, int newSerialisedId) {
...
+	}
+
+	@Override
+	public void forEach(Consumer<? super T> consumer) {
+		values().forEach(consumer);
```
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java]: Same as above

File: src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java [hunk 4]
```diff
@@ -248,4 +246,44 @@ protected void onRemap(T remappedValue, int newSerialisedId) {
...
+	}
+
+	@Override
+	public Spliterator<T> spliterator() {
+		return values().spliterator();
```
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java]: Same as above

**Review by valoeghese — CHANGES_REQUESTED** [review]
Just some code style changes.
I won’t complain about the import order changes since I don’t have a preferred style for that (yet).

**Comment by valoeghese** [comment]
Also can you bump version to 0.3.1 in the gradle properties

File: src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java (hunk 1 above)
Comment (williambl) [reply src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java]: Fixed in 3fa96f1.

File: src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java (hunk 2 above)
Comment (williambl) [reply src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java]: Fixed in 3fa96f1.

File: src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java (hunk 3 above)
Comment (williambl) [reply src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java]: Fixed in 3fa96f1

File: src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java (hunk 4 above)
Comment (williambl) [reply src/main/java/io/github/minecraftcursedlegacy/api/registry/Registry.java]: Fixed in 3fa96f1.

**Review by valoeghese — APPROVED** [review]

**Pull request merged**